### PR TITLE
VIP address feature

### PR DESCRIPTION
**Pull request by shchung**
Added vip Feature to repmgrd.

### CODE REVIEW

File: repmgrd-physical.c [hunk 1]
```diff
@@ -209,6 +276,13 @@ monitor_streaming_primary(void)
 	instr_time	log_status_interval_start;
 	PQExpBufferData event_details;
 
+	/* vip_default_value should be changed */
+	bool    vip_bind = false;
+    char    vip_address[MAXLEN] = "";
```
Comment (befehlen) [inline repmgrd-physical.c]: wrong indentation. you need to use a tab instead of spaces.

File: repmgrd-physical.c [hunk 2]
```diff
@@ -256,6 +332,10 @@ monitor_streaming_primary(void)
 		 */
 		if (is_server_available(local_node_info.conninfo) == false)
 		{
+            if(vip_bind)
```
Comment (befehlen) [inline repmgrd-physical.c]: - wrong indentation
- if_(vip_bind)

File: configfile.h [hunk 3]
```diff
@@ -161,6 +161,10 @@ typedef struct
 
 	/* undocumented test settings */
 	int			promote_delay;
+	/* undocumented test settings */
+	char		vip_address[MAXLEN];
```
Comment (befehlen) [inline configfile.h:165]: It seems like there need to be more indent.

File: repmgrd-physical.c [hunk 4]
```diff
@@ -198,6 +201,70 @@ do_physical_node_check(void)
...
+/*
+ * do vip assign
+ */
+bool
+do_vip_assign(char *vip_address, char *vip_nic, char *vip_path)
+{
+    char    command[MAXLEN] ="";
+    int r;
+
+    if( *config_file_options.vip_address != '\0')
```
Comment (befehlen) [inline repmgrd-physical.c]: if_(*config_file_options.vip_address != '\0')

File: repmgrd-physical.c [hunk 5]
```diff
@@ -198,6 +201,70 @@ do_physical_node_check(void)
...
+    else
+    {
+        return false;
+    }
+
+    snprintf(command, sizeof(command), "%s add %s %s", vip_path, vip_nic , vip_address);
+
+    r = system(command);
+    if (r != 0 ){
+        log_notice(_("Error to bind vip address."));
```
Comment (befehlen) [inline repmgrd-physical.c]: unnecessary white space

File: repmgrd-physical.c [hunk 6]
```diff
@@ -198,6 +201,70 @@ do_physical_node_check(void)
...
+    }
+    else
+    {
+        return false;
+    }
+
+    snprintf(command, sizeof(command), "%s add %s %s", vip_path, vip_nic , vip_address);
+
+    r = system(command);
+    if (r != 0 ){
```
Comment (befehlen) [inline repmgrd-physical.c]: unnecessary white space.

File: repmgrd-physical.c [hunk 7]
```diff
@@ -198,6 +201,70 @@ do_physical_node_check(void)
...
+        log_notice(_("Error to bind vip address."));
+        log_error(_("vip command: %s ."),command);
```
Comment (befehlen) [inline repmgrd-physical.c]: log_error(_("vip command: %s ."),_command);

File: repmgrd-physical.c [hunk 8]
```diff
@@ -198,6 +201,70 @@ do_physical_node_check(void)
...
+        return false;
+    }
+    return true;
+}
+
+bool
+do_vip_resign(char *vip_address, char *vip_nic, char *vip_path)
+{
+    char    command[MAXLEN] ="";
```
Comment (befehlen) [inline repmgrd-physical.c]: char    command[MAXLEN] =_"";

File: repmgrd-physical.c [hunk 9]
```diff
@@ -198,6 +201,70 @@ do_physical_node_check(void)
...
+    int r;
+
+    snprintf(command, sizeof(command), "%s del %s %s", vip_path, vip_nic , vip_address);
+
+    r = system(command);
+    if (r != 0 ){
+        log_notice(_("Error to unbind vip address."));
+        return true;
```
Comment (befehlen) [inline repmgrd-physical.c]: Why do_vip_resign() returns true when failed on binding VIP address.

**Comment by befehlen** [comment]
You need to add descriptions to repmgr.conf.sample.

**Review by befehlen — APPROVED** [review]
+1, Looks good to me, someone needs to approve.

File: vip.sh [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+		exit 1
+	fi
+else
+	echo "Please Check IP Address"
+	exit 1
+fi
+
+# add/del IP Address to Interface
+
+sudo ip addr $TYPE $IP/$PREFIX dev $NIC label $NIC:1
```
Comment (ppagasari) [inline vip.sh:66]: label 을 다는 이유가 따로 있나요?

**Review by ppagasari — APPROVED** [review]

**Pull request merged**